src/mbgl/math/log2.cpp

```diff
@@ -21,15 +21,9 @@ uint32_t ceil_log2(uint64_t x) {
     return y;
 }
 
-double log2(double x) {
-// log2() is producing wrong results on ARMv5 binaries
-// running on ARMv7+ CPUs.
-#if defined(__ANDROID__)
-    return std::log(x) / 0.6931471805599453; // log(x) / log(2)
-#else
-    return ::log2(x);
-#endif
-}
+template <> float log2(float x);
+template <> double log2(double x);
```

Comment:
This accounts for [explicit instantiation](http://en.cppreference.com/w/cpp/language/template_specialization), which from my understanding _instantiates_ these template specializations to be reused by each compilation unit without duplications.